components/Tabs/Tabs.js

```diff
@@ -1,58 +1,70 @@
 class TabLink {
   constructor(element){
     // assign this.element to the element reference
-    this.element;
+    this.element = element;
     // Get the tab data attribute and save the value here
-    this.tabData = ; 
+    this.tabData = this.element.dataset.tab;
+
     // Find all elements with the .card class in index.html that correspond to the tab data attribute
     // If the data is 'all' then select all cards regardless of their data attribute
-    if(){
-      this.cards = ;
+    if(this.tabData === "all"){
+      this.cards = document.querySelectorAll('.card');
+      console.log("test1")
     } else {
-      this.cards = ;
+      this.cards = document.querySelectorAll(`.card[data-tab="${this.tabData}"]`);
+      console.log("test2")
     }
 
-    // Map over the cards array and convert each card element into a new instance of the TabCard class. 
+    // Map over the cards array and convert each card element into a new instance of the TabCard class.
     // Pass in the card object to the TabCard class.
-    this.cards = Array.from(this.cards).map();
+    this.cards = Array.from(this.cards).map(card => new TabCard(card));
     // Add a click event that invokes selectTab
-    this.element.addEventListener();
+    this.element.addEventListener('click',  ()=>{this.selectTab()});
   }
 
   selectTab(){
 
     // Select all elements with the .tab class on them
-    const tabs = document.querySelectorAll();
+    let tabs = document.querySelectorAll('.tab');
+    for (let i=0; i<tabs.length; i++){
```

Comment:
Although for loops do work, let's try to utilize ES6 syntax to make code more readable.